src/client/api/object.c

```diff
@@ -264,3 +265,45 @@ daos_obj_layout_get(daos_handle_t coh, daos_obj_id_t oid,
 
 	return rc;
 }
+
+int
+daos_obj_verify(daos_handle_t coh, daos_obj_id_t oid, daos_epoch_t epoch)
+{
+	tse_task_t	*task;
+	daos_handle_t	 oh;
+	daos_handle_t	 th;
+	daos_epoch_t	*epoch_p = NULL;
+	int		 epoch_nr = 0;
+	int		 rc;
+
+	rc = daos_obj_open(coh, oid, 0, &oh, NULL);
+	if (rc != 0)
+		return rc;
+
+	rc = dc_tx_local_open(coh, epoch, &th);
+	if (rc != 0)
+		goto out;
+
+again:
+	/* Sync object against the given @epoch. */
+	rc = dc_obj_sync_task_create(oh, th, &epoch_p, &epoch_nr,
+				     NULL, NULL, &task);
+	if (rc == 0) {
+		rc = dc_task_schedule(task, true);
+		if (rc == 0) {
+			rc = dc_obj_verify(oh, epoch_p, epoch_nr);
```

Comment:
This looks not correct to me, imagine user wants to verify a snapshot A, and it passed in the HLC of the snapshot as @epoch, but we actually verify the epochs returned by sync here?

That's why I like to have a 'HLC sync (all bump to highest)' on object sync, and always verify single unified epoch like what we did for snapshot.